src/Indicator.vala

```diff
@@ -35,10 +40,15 @@ public class Power.Indicator : Wingpanel.Indicator {
     }
 
     construct {
+        init_bus ();
         popover_widget = new Widgets.PopoverWidget (is_in_session);
-        display_widget = new Widgets.DisplayWidget (popover_widget);
-
         popover_widget.settings_shown.connect (() => close ());
+        popover_widget.update_brightness.connect (on_update_brightness);
+
+        display_widget = new Widgets.DisplayWidget ();
+        display_widget.indicator_scroll.connect ((event)=> {
+            popover_widget.on_scroll_brightness_slider (event);
```

Comment:
perhaps we can harmonize the signal/function names? Right now we have `indicator_scroll`, `on_scroll_brightness_slider`.

Also, you're sending the information to the `popover_widget` via `on_scroll_brightness_slider` only to get it back via `update_brightness`. I think the `DisplayWidget` should process the scroll event by itself and just send a change amount (a float between 0 and 1) to Indicator. Same about `PopoverWidget`. And if you need to share the same handling code in `DisplayWidget` and `PopoverWidget`, just add a static method in Utils.vala.

Reply:
Yes, that's a long journey for `scroll_event`. But I'm also handling `scroll_event` from `image_box` in `BrightnessSlider` and I want to process this scroll event in the same way everywhere. So I'm going to place it in Utils.vala.

Reply:
Yeah. Create a static function in Utils and use that in both places. This way we avoid spaghettiness.